### ISSUE TITLE
iD is too bold in the "Warnings" about potential errors

### ISSUE THREAD

**Issue by dieterdreist** [openstreetmap/iD#11573]
**Describe the bug**
1. Currently, there are only "Warnings" for tags and situations that iD does not know of. This should be fixed. Having several severity classes for this kind of potential problems would be a big step forward. Additionally, a disclaimer should be put above every of such messages and hints, to make it clear that these aren't errors or problems, they are eventually problems, but could also just mean iD has the wrong rules or does not yet know about these tags.

2. Also these warnings and tag corrections are displayed to new mappers who cannot know how to deal with them, and do not have the experience and knowledge to make the decision to "fix" or not act upon these recommendations.

3. Rather than having a single class of "problem", it could be distinguished between "error" (likely an error), "warning"  and "notice".

**Expected behavior**
Be clear and repeat it every time when a "warning" is displayed, that this is not the "truth", but just an automated hint about a potential problem which may well be completely fine.

Do not show any such calls for action to people with less than 100 changesets (or 300 changesets). It only pushes them into taking decisions they cannot make.

DIfferentiate between "error", "warning", "notice".

**Screenshots**
![](https://community-cdn.openstreetmap.org/uploads/default/original/2X/3/3c380adc720222a42d5e9d5614cb10700b608d34.jpeg)

**Additional context**

**Comment by 1ec5** [comment]
iD does differentiate between warnings and errors. An error, such as a feature with no tags, shows up in red and prevents you from saving.

Historically, whereas other editors might implement some of the more pedantic issues as notices, iD would simply omit them from the validator, relying on QA tools such as KeepRight or Osmose to surface them instead. These tools are also integrated into iD but less in your face and more clearly geared toward advanced users.

If you’re concerned about users feeling pressured to resolve warnings without having the necessary context, I think the biggest source of pressure is that generated and ignored warnings currently count against your permanent record in the How Did You Contribute? tool. I don’t think the validator warnings were ever intended to have that kind of accusatory effect. iD’s walkthrough could also more clearly explain the need to be pragmatic about the warnings that appear.

(On the other hand, if you simply disagree with a certain class of issues being implemented as validator warnings versus being left to a QA tool, that would be a different discussion.)

**Comment by matkoniecz** [comment]
Related (all three about NSI): 

https://github.com/openstreetmap/iD/issues/9529 - NSI suggestions are treated as valid when they are often not

https://github.com/openstreetmap/iD/issues/9528 - It is not possible for a mapper to understand why an invalid NSI suggestion was made

https://github.com/openstreetmap/iD/issues/6517 - Make issue warning clearer when offering to add brand:wikidata or brand:wikipedia

**Comment by matkoniecz** [comment]
this looks like something unfixable in this repo, and belonging to https://github.com/openstreetmap/iD

**Comment by matkoniecz** [comment]
> Currently, there are only "Warnings" for tags and situations that iD does not know of. 

what you mean by that?

> DIfferentiate between "error", "warning", "notice".

notice level report was created since then

>  Be clear and repeat it every time when a "warning" is displayed, that this is not the "truth", but just an automated hint about a potential problem which may well be completely fine.

partially handled by #11030 - and we have specific issues for NSI

> Do not show any such calls for action to people with less than 100 changesets (or 300 changesets). It only pushes them into taking decisions they cannot make.

even about newly created ones they created?

**Comment by tyrasd** [comment]
> Rather than having a single class of "problem", it could be distinguished between "error" (likely an error), "warning" and "notice".

Yes, since #11020 there is a dedicated type for less severe warnings. But in order to make this issue actually addressable, we'd need concrete suggestions about which validation warnings are currently sorted into the "wrong" category. Let's work on that as open new issues for specific validations.
